cdap-common/bin/functions.sh

```diff
@@ -643,6 +643,14 @@ cdap_start_java() {
       __defines+=" -Djava.library.path=${JAVA_LIBRARY_PATH}"
     fi
     __startup_checks=${CDAP_STARTUP_CHECKS:-$(cdap_get_conf "master.startup.checks.enabled" "${CDAP_CONF}"/cdap-site.xml true)}
+
+    # Build and upload coprocessor jars
+    logecho "$(date) Ensuring required HBase coprocessors are on HDFS"
+    "${JAVA}" ${JAVA_HEAPMAX} ${OPTS} -cp ${CLASSPATH} co.cask.cdap.data.tools.CoprocessorBuildTool </dev/null >>${__logfile} 2>&1
```

Comment:
Those functions *should* be no-op/idempotent, if the variables are already set. If they're not, it's a bug. It's better to setup the function in a way that we reuse it, versus having the same code in multiple places.